src/ledger/backend/contracts_deposit.spec.ts

```diff
@@ -4,38 +4,58 @@
 import { expect } from "chai";
 
 import { Wallet } from "ethers";
-import { ETHZERO, Amount } from "#erdstall/ledger/assets";
+import { ETHZERO, Amount, Asset } from "#erdstall/ledger/assets";
 import { Address } from "#erdstall/ledger";
 import {
 	makeETHDepositCalls,
 	makeERC20DepositCalls,
 	makeERC721DepositCalls,
 } from "./contracts_deposit";
+import { DepositerCallsFactory } from "./tokenmanager";
+import { PerunArt__factory } from "./contracts/factories/PerunArt__factory";
 
 import { Enviroment, setupEnv } from "#erdstall/test/ledger";
+import * as test from "#erdstall/test";
+
+const TOKEN_SIZE = 4;
 
 describe("Deposit_Call_Wrapper", () => {
+	const rng = test.newPrng();
 	let testenv: Enviroment;
 	let bob: Wallet;
-	const amount = new Amount(100n);
+	const amount = new Amount(test.newRandomBigInt(rng, 16));
+	const tokens = test.newRandomTokens(rng, TOKEN_SIZE);
 
 	before(async () => {
 		testenv = await setupEnv();
 		bob = testenv.users[0];
+
+		const part = PerunArt__factory.connect(testenv.perunArt, bob);
+		for (const id of tokens.value) {
+			await part.mint(bob.address, id);
+		}
 	});
 
 	it("allows deposits for ETH, ERC20 and ERC721", async () => {
 		let nonce = await bob.getTransactionCount();
 
-		[
-			[makeETHDepositCalls, testenv.ethHolder, ETHZERO],
-			[makeERC20DepositCalls, testenv.erc20Holder, testenv.perun],
-		].forEach(async ([call, holder, token]) => {
-			const depCall = (call as typeof makeERC20DepositCalls)(
+		const testCases: [DepositerCallsFactory, string, string, Asset][] = [
+			[makeETHDepositCalls, testenv.ethHolder, ETHZERO, amount],
+			[makeERC20DepositCalls, testenv.erc20Holder, testenv.perun, amount],
+			[
+				makeERC721DepositCalls,
+				testenv.erc721Holder,
+				testenv.perunArt,
+				tokens,
+			],
+		];
```

Comment:
So it's never tested that the `Session` itself implements `deposit` correctly, but only the intermediary functions `makeFooDepositCalls`?

Reply:
Very good. I forgot extending the `LedgerWriter` spec to also deposit `Tokens` :rocket: Added.

Reply:
Which itself is just exposed by the `Session` implementation.

Reply:
Yes, in the end to end test. But it is still missing a erc721 deposit case.

Reply:
The only reason I did not bother adding an extra testcase there, although beneficial in the long run, is because I could not immediately wrap my head around how the end to end test is structured.
And considering the `Session.deposit(...)` function looks like this:

```ts
	async deposit(
		assets: Assets,
	): Promise<Stages<Promise<ethers.ContractTransaction>>> {
		if (!this.erdstallConn) {
			return Promise.reject(ErrUnitialisedClient);
		}

		return (this.erdstallConn as LedgerWriter).deposit(assets);
	}
```

Which is essentially just `this.erdstallConn.deposit(assets)`. And this function is tested already.

This is just the quickshot, I can dig into the end to end test and setup another testcase there.

Reply:
OK thanks, I guess it is fine for now then. Would still be good to add an issue about it though, so we don't forget...

Reply:
Tracked in #86